### PR TITLE
Switch to use callbacks->clusterInfo in built-in filters' code.

### PR DESCRIPTION
**Pull request by stevenzzzz**
For an explanation of how to fill out the fields, please see the relevant section 
in PULL_REQUESTS.md

*Description*: Switch to use callbacks->clusterInfo in built-in filters' code. as PR #4600 gets merged, we can use en/decoderFilterCallbacks_->clusterInfo() to get current routeEntry clusterInfo.

*Risk Level*: LOW(refactor only)
*Testing*: unit test.
*Docs Changes*:
*Release Notes*: Switch to use callbacks->clusterInfo in built-in filters' code. 
*Fixed Issue*: #4643

### CODE REVIEW

**Comment by stevenzzzz** [comment]
Interesting, didn't know there is this funcion. So glad now we have a uniformed way to access clusterInfo.
I will update the PR.

**Comment by stevenzzzz** [comment]
PR updated, seems all tests happy.

File: source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace GrpcHttp1Bridge {
  */
 class Http1BridgeFilter : public Http::StreamFilter {
 public:
-  Http1BridgeFilter(Upstream::ClusterManager& cm) : cm_(cm) {}
+  Http1BridgeFilter() {}
```
Comment (alyssawilk) [inline source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.h]: Can we kill off the empty constructors entirely here and below?

Comment (stevenzzzz) [reply source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.h]: Done.
I thought leave it along might be more readable, bad taste.

Comment (alyssawilk) [reply source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.h]: Nah, it's just personal style.  Probably someone else on the project prefers them in as well :-P

**Comment by alyssawilk** [comment]
@lizan since a few of these filters are yours, you Ok doing the cross company ( :-( ) sign-off?

**Pull request merged**